packages/commerce-sdk-react/src/hooks/useAsync.ts

```diff
@@ -24,21 +25,27 @@ export const useAsync = <T>(fn: () => Promise<T>, deps?: unknown[]): QueryRespon
     return result
 }
 
-export const useAsyncExecute = <A, R>(fn: (arg: A) => Promise<R>) => {
+export const useAsyncExecute = <A extends any[], R>(fn: (...args: A) => Promise<R>) => {
     // This is a stub implementation to validate the types.
     // The real implementation will be more React-y.
+    const [isLoading, setIsLoading] = useState<boolean>(false)
+    const [data, setData] = useState<R | undefined>(undefined)
+    const [error, setError] = useState<Error | undefined>(undefined)
     const result: ActionResponse<A, R> = {
-        isLoading: false,
-        execute(arg: A) {
-            result.isLoading = true
-            fn(arg)
-                .then((data) => {
-                    result.isLoading = false
-                    result.data = data
+        data,
+        error,
+        isLoading,
+        execute: (...arg) => {
+            setIsLoading(true)
+            fn(...arg)
```

Comment:
can we use await/async syntax instead of promise chain? 🤔

Reply:
We technically could, but I think that a promise chain makes more sense in this specific instance. Since React doesn't play nice with promises, I think we want `execute` to remain a sync method. Which means that using async/await would require an async [iife](https://developer.mozilla.org/en-US/docs/Glossary/IIFE), which seems like extra boilerplate for three lines of code. (Also, typescript-eslint has a [no-floating-promises rule](https://typescript-eslint.io/rules/no-floating-promises/) that an async IIFE would violate, but this promise chain does not.)

Reply:
How about this? 
```
execute: () => {
  setIsLoading(true)
  const runCallback = async () => {
    try {
       const res = fn(...arg)
       setData(res)
       setIsLoading(false)
    } catch (error) {
      setIsLoading(false)
     setError(error)
    }
  }
  runCallback()
}
```